### ISSUE TITLE
[BUG]  Searching for closest structure among log message

### ISSUE THREAD

**Issue by Cheekk1** [ChoiceTheorem/ChoiceTheorem-s-overhauled-village#219]
Faced an issue with performance on server, which is related to this message:

> [19:59:34] [pool-10-thread-4/INFO]: Searching for closest structure among [ResourceKey[minecraft:worldgen/structure / ctov:large/village_swamp], ResourceKey[minecraft:worldgen/structure / ctov:medium/village_beach], ResourceKey[minecraft:worldgen/structure / ctov:large/village_jungle], ResourceKey[minecraft:worldgen/structure / ctov:medium/village_taiga_fortified], ResourceKey[minecraft:worldgen/structure / ctov:small/village_jungle], ResourceKey[minecraft:worldgen/structure / ctov:large/village_desert_oasis], ResourceKey[minecraft:worldgen/structure / ctov:small/village_desert], ResourceKey[minecraft:worldgen/structure / ctov:small/village_savanna_na]] from pos class_2338{x=341, y=103, z=917}
> [19:59:40] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 3476ms or 69 ticks behind

> [00:27:51] [Worker-Main-186/WARN]: Non-existent template pool reference: ctov:village/bounty/plains
> [00:27:51] [Worker-Main-186/WARN]: Non-existent template pool reference: ctov:village/vampirism/totem
> [00:27:51] [Worker-Main-186/WARN]: Non-existent template pool reference: ctov:village/waystone/normal
> [00:28:04] [Worker-Main-148/WARN]: Non-existent template pool reference: ctov:village/vampirism/totem
> [00:28:04] [Worker-Main-148/WARN]: Non-existent template pool reference: ctov:village/waystone/normal
> [00:28:04] [Worker-Main-148/WARN]: Non-existent template pool reference: ctov:village/bounty/plains

Any idea which files are missing, and how i can fix that ?

Fabric 1.20.1
https://www.curseforge.com/minecraft/modpacks/bcg modpack with additionally installed CTOV + Wabi-Sabi Structures mods

**Comment by ChoiceTheorem** [comment]
Did you use any command before this?

**Comment by Cheekk1** [comment]
No. 2 players were on the server, map wasnt preloaded (or pre generated).

**Comment by Korenchkin** [comment]
i will add to this,i have updated latest on 1.21.1,i have about 250 mods(my choice only,no official modpack,basically just mods,no configs) and thanks to bluemap i pinpointed the problem to those lamp posts possibly with arrows for nearest villages
`Searching for closest structure among [ResourceKey[minecraft:worldgen/structure / ctov:small/village_savanna], ResourceKey[minecraft:worldgen/structure / ctov:large/village_savanna], ResourceKey[minecraft:worldgen/structure / ctov:large/village_desert_oasis], ResourceKey[minecraft:worldgen/structure / ctov:small/village_christmas], ResourceKey[minecraft:worldgen/structure / ctov:medium/village_mushroom], ResourceKey[minecraft:worldgen/structure / ctov:small/village_jungle_tree], ResourceKey[minecraft:worldgen/structure / ctov:medium/village_mountain], ResourceKey[minecraft:worldgen/structure / ctov:medium/village_swamp_fortified]]: from pos BlockPos{x=-597, y=81, z=37}`
at those coords
not sure if same as bug #239 ,so adding it here...
might help?
